README.md

```diff
@@ -13,10 +13,17 @@
 
 ## Table of Contents
 
+- [Purpose](#purpose)
 - [Building](#building)
 - [Getting started](#getting-started)
 - [Next steps](#next-steps)
 
+## Purpose and terminology
+
+The purpose of the tool contained in this repository is to provide means to manipulate the intermediate representation (IR) of quantum programs. We refer to these IRs as quantum IRs (QIRs). The main tool provided here is QIR adaptor tool (QAT) which allows one to transform one QIR to another QIR that fulfils certain requirements.
```

Comment:
The line length here greatly exceeds that allowed by the current markdownlint configuration; it may be worth either breaking it up or setting markdownlint to disable line length checks so as to reduce extraneous warnings.